frontend/pkg/frontend/middleware_logging.go

```diff
@@ -71,13 +73,45 @@ func MiddlewareLogging(w http.ResponseWriter, r *http.Request, next http.Handler
 	logger = logger.With(
 		"request_method", r.Method,
 		"request_path", r.URL.Path,
+	)
+
+	// make a best attempt at parsing the resourceID. This will often fail because we have non-resource requests.
+	// we do this so that we can add subscription, resourceGroup, and hcpCluster to the logger context for future searching
+	// if possible.
+	// It's important to do before the second panic handler so that panics can be correlated easily.
+	// TODO are the value we find case sensitive or case insensitive.  They used to be case sensitive, so I have left that
+	if resourceID, err := azcorearm.ParseResourceID(r.URL.Path); err == nil {
+		logger = logger.With(
+			"subscription_id", resourceID.SubscriptionID,
+			"resource_group", resourceID.ResourceGroupName,
+		)
+
+		currID := resourceID
+		for currID != nil {
+			// TODO we have the option on recording each type.  I have no real preference
+			if currID.ResourceType.String() == strings.ToLower(api.ClusterResourceType.String()) {
+				logger.With(
+					"hcp_cluster_name", currID.Name,
+				)
+				break
+			}
+			currID = currID.Parent
+		}
+	}
+
+	// include the context values (logger.With) with every line so we can grep for them.
+	ctx = ContextWithLogger(ctx, logger)
+	r = r.WithContext(ctx)
+
+	logger.Info("read request",
 		"request_proto", r.Proto,
 		"request_query", r.URL.RawQuery,
+		// TODO referrer is under the client's control.  Printing it out could be huge.
```

Comment:
It's actually not.  ARM provides the referer header to RPs, overriding the client's value if present.